### ISSUE TITLE
Not everything from colorscheme is colored.

### ISSUE THREAD

**Issue by ftpd** [macvim-dev/macvim#1111]
**Describe the bug**
When I open the same file in builtin vim and MacVim, the latter doesn't color everything.

**To Reproduce**
1. Open file with /usr/bin/vim
2. Open the same file vim MacVim

**Expected behavior**
The same colors in both instances.

**Screenshots**
https://imgur.com/a/fu0BGoB

**Environment (please complete the following information):**
 - Vim version: builtin 8.2.850, MacVim 8.2.1719
 - OS: macOS 11.0 Big Sur (Beta 10)
 - Terminal: iTerm2 3.3.12

**Comment by ychin** [comment]
@ftpd Can you provide more information? It's not very helpful to just have a screenshot to look at without any more relevant information to go from. Have you tried doing `vim --clean` and see if the issue is there? Also, what doe `echo $TERM` show in the terminal, and what colorscheme are you using?

**Comment by ychin** [comment]
Also, just to make sure I understand, you are opening MacVim in terminal right?

**Comment by ftpd** [comment]
Sure, I will provide every information you need.
- my TERM is xterm-256color;
- I will continue using the word 'builtin' for the default vim shipped with my OS, the one located in /usr/bin/vim - I hope that's ok;
- vim --clean, for both 'builtin' (invoked via /usr/bin/vim --clean <filename> and MacVim (invoked by mvim --clean <filename>, shows two different 'ways of colorize' (sorry, I don't know the better word and I'm not a native English speaker) of the file, although on both colorscheme is set to 'default' - for example in 'set title' line 'set' is yellow on both, but 'title' string is blue in builtin and magenta in MacVim;
- colorscheme is https://github.com/morhetz/gruvbox;
- for test I'm using the same file (my personal vimrc, available (here)[https://github.com/ftpd/dotfiles/blob/master/vim/vimrc];
- both vims are using this vimrc file for configuration, so it's the same vim-plug and the same colorscheme, loaded from one copy on disk, not two separate copies lying in two separated places;
- it doesn't matter if I invoke MacVim from terminal (by bundled 'mvim' script) or from Spotlight (or even clicking app's icon in /Applications) and doing :e <filename>. The result is the same.

Basically, just to be clear: I don't care than colors in MacVim are slightly different, like cyan is more 'vivid' or something. Please just take a look on the lines starting with 'Plug' on my screenshot. In builtin vim strings in ' are green in builtin, and white (or 'the same as normal text, not colorized at all') in MacVim.

Two observations:
- when I start MacVim inside the terminal (mvim -v), it looks just like the gui version, so not colorizing Plug '<string>';
- when I start both vims with --clear, so with 'default' colorscheme, builtin vim also highlight those strings (but on different color, obviously, as it's not set by external colorscheme), and MacVim doesn't.

Let me provide one more screenshot ;-), just in case I didn't explain everything properly. Here is "/usr/bin/vim --clean .vimrc" on default 'Dark background' colorscheme from iTerm (bigger window) and "mvim --clean .vimrc." in MacVim GUI (smaller window with smaller font (iTerm settings showing colorscheme just for convenience).
https://imgur.com/a/jHOxwqW

**Comment by ftpd** [comment]
Ok, next clue: it might be actually vim itself's fault. I've tried on 8.2.1767 on my Ubuntu VPS and on self-compiled 8.2.1862 on my work CentOS 7 (don't ask...) and in both of them the result is the same as in MacVim with 8.2.1719... I will try older build maybe?

**Comment by ftpd** [comment]
Yeah, that's the case. I've tried MacVim build 163 (vim 8.2.539) and it looks like in my builtin 8.2.850, so it colors Plug 'blablabla' line; and snapshot 164 (vim 8.2.1456) doesn't.
Something has changed in the upstream between patchiest 850 and 1456 (I don't want/need to know, what exactly and in which version) and it's causing those differences.

@ychin feel free to close this issue - I'm leaving it open just for you to check out my 'solution', as I find it somewhat interesting  ;-)

**Comment by ychin** [comment]
Oh, thanks for doing the treasure hunt! Just out of curiosity I ran a bisect and it was this commit (https://github.com/vim/vim/commit/2547aa930b59f5e2bcb70e81d5a57ed461e59b4f) right after 8.2.1299 that introduced the behavior.

The change introduced a new syntax item called `vimUsrCmd` which includes the whole line where you are using a user-defined command (e.g. `Plug`). Note that when you call `Plug "something/something.vim"`, the string isn't really a Vim string. You will notice if you do `let someVar = "some string"`, the string there will still be colored the old way.

Anyway, closing the issue. Glad we found the cause.
